### PR TITLE
Matching simulation improvements

### PR DESCRIPTION
**Pull request by natemort**
Fix sync matching by increasing the deadline for AddDecisionTask. Sync matching subtracts 1 second from the deadline to determine how long to spend sync matching, and with a deadline of 0.5 seconds it immediately times out.

Add support for parameterizing the configuration file to use. This will allow us to have several scenarios that we evaluate any matching changes against.

Rather than running for a fixed time, run until the expected number of tasks have been consumed. This allows for the benchmark to be used as a measure of throughput and also makes it run faster for short lived scenarios.

**What changed?**
- Add additional options to the matching simulator

**Why?**
- Allows us to better test changes to matching.

**How did you test it?**
- ./scripts/run_matching_simulator.sh
- ./scripts/run_matching_simulator.sh burst

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: host/matching_simulation_test.go [hunk 1]
```diff
@@ -176,31 +181,35 @@ func (s *MatchingSimulationSuite) TestMatchingSimulation() {
 	numPollers := getNumPollers(s.testClusterConfig.MatchingConfig.SimulationConfig.NumPollers)
 	pollDuration := getPollDuration(s.testClusterConfig.MatchingConfig.SimulationConfig.PollTimeout)
 	polledTasksCounter := int32(0)
+	maxTasksToGenerate := getMaxTaskstoGenerate(s.testClusterConfig.MatchingConfig.SimulationConfig.MaxTaskToGenerate)
+	var tasksToReceive sync.WaitGroup
+	tasksToReceive.Add(maxTasksToGenerate)
```
Comment (taylanisikdemir) [inline host/matching_simulation_test.go:186]: Simulation runs for 180s at most ([timeout set here](https://github.com/uber/cadence/blob/06e5a6dc9ec0715298de27e6024234fcddf47656/docker/buildkite/docker-compose-local-matching-simulation.yml#L30)) and `maxTasksToGenerate` might not be hit if it's set too high. please add a comment in the test code here about this to help identify "simulation is stuck" cases due to misconfiguration

**Review by taylanisikdemir — APPROVED** [review]

File: host/testdata/matching_simulation_burst.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  nummatchinghosts: 4
+  simulationconfig:
+    tasklistwritepartitions: 2
+    tasklistreadpartitions: 2
+    numpollers: 10
+    numtaskgenerators: 2
+    taskgeneratortickinterval: 10ms
+    maxtasktogenerate: 1500
+    polltimeout: 5s
+    forwardermaxoutstandingpolls: 20
```
Comment (Shaddoll) [inline host/testdata/matching_simulation_burst.yaml:18]: the default setting in prod is 1

Comment (natemort) [reply host/testdata/matching_simulation_burst.yaml:18]: I'm just adding this as an example for now, I'll be adding a full suite that we can use to validate it.

File: host/matching_simulation_test.go (hunk 1 above)
Comment (natemort) [reply host/matching_simulation_test.go:186]: Done, thanks.

**Pull request merged**